src/py/aspen/app.py

```diff
@@ -99,14 +102,47 @@ def logout():
 @application.route("/api/usergroup", methods=["GET"])
 @requires_auth
 def usergroup():
+
     with session_scope(application.config["DATABASE_CONFIG"].INTERFACE) as db_session:
-        # since authentication is required to access view this information is guaranteed to be in session
         profile = session["profile"]
-        user = (
-            db_session.query(User)
-            .options(joinedload(User.group))
-            .filter(User.auth0_user_id == profile["user_id"])
-            .one()
-        )
+        user = get_usergroup(db_session, profile["user_id"])
         user_groups = {"user": user.to_dict(), "group": user.group.to_dict()}
         return jsonify(user_groups)
+
+
+@application.route("/api/samples", methods=["GET"])
+@requires_auth
+def samples():
+
+    with session_scope(application.config["DATABASE_CONFIG"].INTERFACE) as db_session:
+        profile = session["profile"]
+        user = get_usergroup(db_session, profile["user_id"])
+
+        sequencing_reads = (
+            db_session.query(SequencingReads)
+            .join(Sample, Group, User)
+            .options(
+                joinedload(SequencingReads.sample).joinedload(Sample.submitting_group)
+            )
+            .filter(Sample.submitting_group == user.group)
+            .all()
+        )
```

Comment:
You could plausibly just do:

```suggestion
        sequencing_reads = (
            db_session.query(SequencingReads)
            .join(Sample, Group, User)
            .options(
                joinedload(SequencingReads.sample).joinedload(Sample.submitting_group)
            )
            .filter(User.auth0_user_id == profile["user_id"])
        )
```

Reply:
I don't think you pull the group info, so the joined load to the group is unnecessary.